### ISSUE TITLE
Fragment called twice after orientation change

### ISSUE THREAD

**Issue by philbring** [neokree/MaterialNavigationDrawer#86]
1) add fragment so section
 MaterialSection section = newSection("Info", new MyFragment());
addSection(section);

2) select section

3) change display orientation (rotate device)
Fragment called twice. First tiem with savedInstanceState != null and second time with savedInstanceState = null.

Tested with 1.2.9

The probelem is described here for default *NavigationDrawer*:
http://blog.sqisland.com/2014/06/navigationdrawer-creates-fragment-twice.html

I think you have to modify at least the following method in your *MaterialNavigationDrawer* class (line 585):
**private void setFragment(Fragment fragment,String title,Fragment oldFragment)**

**Comment by philbring** [comment]
Patch for *MaterialNavigationDrawer*

```
--- MaterialNavigationDrawer.java.orig	2015-01-05 20:15:36.000000000 +0100
+++ MaterialNavigationDrawer.java	2015-01-07 12:43:12.000000000 +0100
@@ -457,7 +457,7 @@
         title = section.getTitle();
         currentSection = section;
         section.select();
-        setFragment((Fragment) section.getTargetFragment(), section.getTitle(), null);
+        setFragment((Fragment) section.getTargetFragment(), section.getTitle(), null, savedInstanceState);
 
         // learning pattern
         if(learningPattern) {
@@ -582,7 +582,7 @@
         return currentSection;
     }
 
-    private void setFragment(Fragment fragment,String title,Fragment oldFragment) {
+    private void setFragment(Fragment fragment,String title,Fragment oldFragment, Bundle savedInstanceState) {
         // si setta il titolo
         setTitle(title);
         if(Build.VERSION.SDK_INT < Build.VERSION_CODES.HONEYCOMB) {
@@ -591,7 +591,8 @@
             if(oldFragment != null && oldFragment != fragment)
                 ft.remove((android.support.v4.app.Fragment) oldFragment);
 
-            ft.replace(R.id.frame_container, (android.support.v4.app.Fragment) fragment).commit();
+            if(savedInstanceState == null)
+                ft.replace(R.id.frame_container, (android.support.v4.app.Fragment) fragment).commit();
         }
         else if (fragment instanceof android.app.Fragment) {
             if (oldFragment instanceof android.support.v4.app.Fragment)
@@ -602,7 +603,8 @@
             if (oldFragment != null && fragment != oldFragment)
                 ft.remove((android.app.Fragment) oldFragment);
 
-            ft.replace(R.id.frame_container, (android.app.Fragment) fragment).commit();
+            if(savedInstanceState == null)
+                ft.replace(R.id.frame_container, (android.app.Fragment) fragment).commit();
         }
         else if(fragment instanceof android.support.v4.app.Fragment) {
             if(oldFragment instanceof android.app.Fragment)
@@ -612,7 +614,8 @@
             if(oldFragment != null && oldFragment != fragment)
                 ft.remove((android.support.v4.app.Fragment) oldFragment);
 
-            ft.replace(R.id.frame_container, (android.support.v4.app.Fragment) fragment).commit();
+            if(savedInstanceState == null)
+                ft.replace(R.id.frame_container, (android.support.v4.app.Fragment) fragment).commit();
         }
         else
             throw new RuntimeException("Fragment must be android.app.Fragment or android.support.v4.app.Fragment");
@@ -819,7 +822,7 @@
 
         switch (section.getTarget()) {
             case MaterialSection.TARGET_FRAGMENT:
-                setFragment((Fragment) section.getTargetFragment(), section.getTitle(), (Fragment) currentSection.getTargetFragment());
+                setFragment((Fragment) section.getTargetFragment(), section.getTitle(), (Fragment) currentSection.getTargetFragment(), null);
 
                 changeToolbarColor(section);
                 break;
```

**Comment by mkosik** [comment]
@neokree solution from @philbring seems good to me.

**Comment by muriloandrade** [comment]
@neokree will this be commited in the next release? 
This issue is causing extra fragments added to my FragmentManager, and some classes methods are acting on non visible fragments.

**Comment by rburgst** [comment]
whats the status of this? I seem to have the same problem in 1.3.3

**Comment by pkliang** [comment]
It turns out this bug has been fixed in 1.3.3
